src/Auth/TokenAuthenticator.ts

```diff
@@ -33,34 +37,20 @@ export class Class implements Authenticator<UsernameAndPasswordToken> {
 	authenticate( authenticationToken:UsernameAndPasswordToken ):Promise<Token.Class>;
 	authenticate( credentials:Token.Class ):Promise<Token.Class>;
 	authenticate( authenticationOrCredentials:any ):Promise<Token.Class> {
-		if( Token.Factory.is( authenticationOrCredentials ) ) {
-
-			if( Utils.isString( authenticationOrCredentials.expirationTime ) )
-				authenticationOrCredentials.expirationTime = new Date( <any> authenticationOrCredentials.expirationTime );
-			this._credentials = authenticationOrCredentials;
-
-			return new Promise<Token.Class>( ( resolve:Function, reject:Function ) => {
-				if( ! this.isAuthenticated() ) {
-					this.clearAuthentication();
-					throw new Errors.IllegalArgumentError( "The token provided in not valid." );
-				}
-				resolve( this._credentials );
-			} );
+		if( authenticationOrCredentials instanceof UsernameAndPasswordToken ) return this.basicAuthenticator.authenticate( authenticationOrCredentials ).then( () => {
```

Comment:
If the token was created without using the class constructor, this code will fail.